pkg/persistence/disk_persistence_test.go

```diff
@@ -110,6 +114,176 @@ func TestDiskPersistence_RefuseSave(t *testing.T) {
 	cleanup()
 }
 
+func TestDiskPersistence_Snapshot(t *testing.T) {
+	diskHandle, _ := NewDiskHandle(dataDir)
+	bytesToTest := []byte{115, 111, 109, 101, 10}
+
+	counter := 0
+	diskHandle.(*diskPersistence).snapshotSuffixGenerator = func() string {
```

Comment:
I don't love how we've made the suffix generator mutable here, though I see the use for testing. Wish we had a better way to do this, especially the collision test later on.

Reply:
Yeah, agree. But a kind of compromise was needed here.